docs/guides/contributing.md

```diff
@@ -0,0 +1,429 @@
+
+## Developing Locally
+
+This section will walk you through setting up the [main DoneJS repository](https://github.com/donejs/donejs) on your computer. Remember that DoneJS is [split into multiple repositories](#section=section_ProjectOrganization), but you can apply the same general steps to set up any of the other repos.
+
+### Signing up for GitHub
+
+If you don’t already have a GitHub account, you’ll need to [create a new one](https://help.github.com/articles/signing-up-for-a-new-github-account/).
+
+### Forking & cloning the repository
+
+A “fork” is a copy of a repository in your personal GitHub account. “Cloning” is the process of getting the repository’s source code on your computer.
+
+GitHub has a guide for [forking a repo](https://help.github.com/articles/fork-a-repo/). To fork DoneJS, you can start by going to its [fork page](https://github.com/donejs/donejs/fork).
+
+Next, you’ll want to clone the repo. [GitHub’s cloning guide](https://help.github.com/articles/cloning-a-repository/) explains how to do this on Linux, Mac, or Windows.
+
+GitHub’s guide will [instruct you](https://help.github.com/articles/fork-a-repo/#step-2-create-a-local-clone-of-your-fork) to clone it with a command like:
+
+```shell
+git clone https://github.com/YOUR-USERNAME/donejs
+```
+
+Make sure you replace `YOUR-USERNAME` with your GitHub username.
+
+### Installing the dependencies
+
+After you’ve [forked & cloned the repository](#section=section_Forking_cloningtherepository), you’ll need to install the project’s dependencies.
+
+First, make sure you’ve [installed Node.js and npm](https://docs.npmjs.com/getting-started/installing-node).
+
+If you just cloned the repo from the command line, you’ll want to switch to the folder with your clone:
+
+```shell
+cd donejs
+```
+
+Next, install the project’s dependencies with npm:
+
+```shell
+npm install
+```
+
+### Running the tests
+
+Firefox is used to run the repository’s tests. If you don’t already have it, [download Firefox](https://www.mozilla.org/en-US/firefox/new/). Mozilla has guides for installing it on [Linux](https://support.mozilla.org/t5/Install-and-Update/Install-Firefox-on-Linux/ta-p/2516), [Mac](https://support.mozilla.org/t5/Install-and-Update/How-to-download-and-install-Firefox-on-Mac/ta-p/3453), and [Windows](https://support.mozilla.org/t5/Install-and-Update/How-to-download-and-install-Firefox-on-Windows/ta-p/2210).
+
+After Firefox is installed, you can run:
+
+```shell
+npm test
+```
+
+If every test passed, congratulations! You have everything you need to change the code and have the core team review it.
+
+### Building the documentation
+
+The [main DoneJS repo](https://github.com/donejs/donejs) contains [DoneJS.com](https://donejs.com/).
+
+To build the site, run:
+
+```shell
+npm run document
```

Comment:
[I think it is](https://github.com/donejs/donejs/blob/major/package.json#L11), but if it’s not, I’ll update it when I submit a PR to merge `master` to `major`.